snowshu/adapters/source_adapters/snowflake_adapter.py

```diff
@@ -103,6 +103,100 @@ def _get_all_schemas(self, database: str, exclude_defaults: Optional[bool] = Fal
+        """
+        corrected_database, corrected_name = (
+            self._correct_case(x) for x in (database, name))
+        try:
+            logger.debug("Creating a schema %s in %s database...",
+                         corrected_name, corrected_database)
+            query = f'''CREATE SCHEMA IF NOT EXISTS
+                    {corrected_database}.{corrected_name}'''
+            result = self._safe_query(query)
+            logger.info("Schema creation result: %s", result['status'][0])
+        except ValueError as err:
+            error_message = (
+                f"An error occurred while creating the schema {corrected_name} "
+                f"in database {corrected_database}: {err}"
+            )
+            logger.error(error_message)
+
+    def drop_schema(self, name: str, database: str = 'SNOWSHU'):
+        """Droop a schema and all of its contained objects (tables, views,
+        stored procedures)
+
+            Args:
+                name: The name of the schema to drop
+                database: The database name where the schema is located.
+                            Defaults to SNOWSHU.
+        """
+        corrected_database, corrected_name = (
+            self._correct_case(x) for x in (database, name))
+        try:
+            logger.debug("Creating a schema %s in %s database...",
+                         corrected_name, corrected_database)
+            query = f'''DROP SCHEMA IF EXISTS
+                    {corrected_database}.{corrected_name}
+                    CASCADE'''
+            result = self._safe_query(query)
+            logger.info("Schema drop result: %s", result["status"][0])
+        except ValueError as err:
+            error_message = (
+                f"An error occurred while dropping the schema {corrected_name} "
+                f"in database {corrected_database}: {err}"
+            )
+            logger.error(error_message)
+
+    def create_table(self, query: str, name: str, schema: str, database: str = 'SNOWSHU'):
+        corrected_name, corrected_schema, corrected_database = (
+            self._correct_case(x) for x in (name, schema, database)
+        )
+        full_query = f'''CREATE TABLE IF NOT EXISTS
+            {corrected_database}.{corrected_schema}.{corrected_name}
+            AS {query}'''
+        try:
+            logger.debug("Creating table %s in %s.%s...",
+                         corrected_name, corrected_schema, corrected_database)
+            result = self._safe_query(full_query)
+            logger.info("Table creation result: %s", result['status'][0])
+        except ValueError as err:
+            error_message = (
+                f"An error occurred while creating the table {corrected_name} "
+                f"in database {corrected_database}: {err}")
+            logger.error(error_message)
+
+    def drop_table(self, name: str, schema: str, database: str = 'SNOWSHU'):
+        corrected_name, corrected_schema, corrected_database = (
```

Comment:
I am curios about the specific use case for this function.

Reply:
As I'm still deep diving and exploring the entire idea of snowshu project and how it all works together I think I might not know exactly correct answer for that. Nevertheless I was thinking about it also, why we might need it, if we have `drop_schema w/cascade`, but I decided that for now I'm gonna follow ticket specs, cause I assumed that there might be a possibility that in the future if we notice some discrepancy between the state of a current replica and current config, we will want to drop just the removed table and (if specified) add a new one, without dropping whole schema and rebuilding from scratch. What do you think about it? Does that make sense?